### PR TITLE
check tribe codes when enabling AI/AN verif_type

### PR DESCRIPTION
**Pull request by jacobkagon**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [x] Tests for the changes have been added (for bugfixes / features)

# PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update version)

# What is the ticket # detailing the issue?

Ticket: [ME-186050736](https://www.pivotaltracker.com/n/projects/2640060/stories/186050736)

# A brief description of the changes

Current behavior: If a consumer has AI/AN status and their tribe is located in Maine their AI/AN verification type becomes deactivated every time the consumer role is saved. 

New behavior: The consumer's AI/AN status will not be deactivated and checks to see if the consumer's AI/AN status will be consistent.

# Feature Flag: INDIAN_ALASKAN_TRIBE_CODES_IS_ENABLED

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable that is used to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [x] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

### CODE REVIEW

File: app/controllers/concerns/vlp_doc.rb [hunk 1]
```diff
@@ -98,7 +98,23 @@ def sensitive_info_changed?(role)
...
-    role.person.send("tribal_id") != params_hash["tribal_id"]
+
+    if EnrollRegistry.feature_enabled?(:indian_alaskan_tribe_codes)
+      tribal_state = params.dig("person", "tribal_state")
+      tribe_codes =  params.dig("person", "tribe_codes")
+      tribal_name = params.dig("person", "tribal_name")
+      tribal_id = params.dig("person", "tribal_id")
+
+      if tribal_state == EnrollRegistry[:enroll_app].setting(:state_abbreviation).item
+        role.person.send("tribe_codes") != tribe_codes
```
Comment (vkghub) [inline app/controllers/concerns/vlp_doc.rb:109]: can this be changed to 
```suggestion
        role.person.tribe_codes != tribe_codes
```

File: app/controllers/concerns/vlp_doc.rb [hunk 2]
```diff
@@ -98,7 +98,23 @@ def sensitive_info_changed?(role)
...
+      elsif tribal_id.present?
+        role.person.send("tribal_id") != tribal_id
```
Comment (vkghub) [inline app/controllers/concerns/vlp_doc.rb:111]:
```suggestion
        role.person.tribal_id != tribal_id
```

File: app/controllers/concerns/vlp_doc.rb [hunk 3]
```diff
@@ -98,7 +98,23 @@ def sensitive_info_changed?(role)
...
+      else
+        role.person.send("tribal_name") != tribal_name
```
Comment (vkghub) [inline app/controllers/concerns/vlp_doc.rb:113]:
```suggestion
        role.person.tribal_name != tribal_name
```

File: app/controllers/concerns/vlp_doc.rb [hunk 4]
```diff
@@ -98,7 +98,23 @@ def sensitive_info_changed?(role)
...
+      end
+    else
+      params_hash = params.permit("tribal_id").to_h
+      role.person.send("tribal_id") != params_hash["tribal_id"]
```
Comment (vkghub) [inline app/controllers/concerns/vlp_doc.rb:117]:
```suggestion
      role.person.tribal_id != params_hash["tribal_id"]
```

**Review by vkghub — CHANGES_REQUESTED** [review]

**Review by vkghub — APPROVED** [review]
LGTM

**Review by jayreddy519 — APPROVED** [review]
lgtm..!! 
additional check on tribe code to enable AI/AN verification type.

**Pull request merged**